uk.co.compucorp.civicrm.hrleaveandabsences/tests/phpunit/helpers/SessionHelpersTrait.php

```diff
@@ -11,4 +11,8 @@ private function unregisterCurrentLoggedInContactFromSession() {
     $session = CRM_Core_Session::singleton();
     $session->set('userID', null);
   }
+
+  private function addPermissions(Array $permissions = []) {
+    CRM_Core_Config::singleton()->userPermissionClass->permissions = $permissions;
```

Comment:
This is not how I would expect this method to work. Given the `add` prefix, I would assume that, in case I already have some permissions set, it would just add the given ones instead of replacing it. Something more like this:

```php 
private function addPermission($permission) {
   if (!is_array(CRM_Core_Config::singleton()->userPermissionClass->permissions)) {
     CRM_Core_Config::singleton()->userPermissionClass->permissions = [];
   }
   CRM_Core_Config::singleton()->userPermissionClass->permissions[] = $permission;
}
```